src/sidebars/favourites/favourite-controller.js

```diff
@@ -129,25 +131,25 @@
 
                             // Repeat the check as in the mean time a stock for this favourite could have been added.
                             if (this.stocks.map((stock1) => { return stock1.code; }).indexOf(favourite) === -1) {
-                                var data = stock.data[0],
+                                var data = stock && stock.data && stock.data[0],
```

Comment:
I'm surprised that this works. Doesn't this set `data` to a boolean? How then are things like `data.close` accessed later?